addons/payment/models/payment_acquirer.py

```diff
@@ -193,15 +193,27 @@ def _ensure_payment_method_line(self, allow_create=True):
                 limit=1,
             )
         if pay_method_line:
+            pay_method_line.name = self.name
             pay_method_line.payment_acquirer_id = self
             pay_method_line.journal_id = self.journal_id
         elif allow_create:
             default_payment_method_id = self._get_default_payment_method_id()
-            self.env['account.payment.method.line'].create({
+            create_values = {
+                'name': self.name,
                 'payment_method_id': default_payment_method_id,
                 'journal_id': self.journal_id.id,
                 'payment_acquirer_id': self.id,
-            })
+            }
+            pay_method_line_same_code = self.env['account.payment.method.line'].search(
+                [
+                    ('journal_id.company_id', '=', self.company_id.id),
```

Comment:
```suggestion
                    ('company_id', '=', self.company_id.id),
```

https://github.com/odoo/odoo/blob/471c9753547dffa596ed4f689a39b7f52474a560/addons/account/models/account_payment_method.py#L119